lib/apartment/elevators/subdomain.rb

```diff
@@ -38,13 +43,23 @@ def subdomain(host)
       end
 
       def subdomains(host)
-        return [] unless named_host?(host)
+        host_valid?(host) ? parse_host(host) : []
+      end
+
+      def host_valid?(host)
+        not ip_host?(host) and domain_valid?(host)
+      end
+
+      def ip_host?(host)
+        !/^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$/.match(host).nil?
```

Comment:
The idea here wasn't to validate an IP but just to check if it's an IP... If it's valid or not I don't think it matters ...